### ISSUE TITLE
Fix panic if no connection is accepted for deadline set.

### ISSUE THREAD

**Issue by autholykos** [dusk-network/dusk-blockchain#353]
This will panic if no connection is accepted for deadline set.

```go
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4f8ed9]

goroutine 7 [running]:
net.(*conn).ok(...)
        /usr/local/go/src/net/net.go:175
net.(*conn).RemoteAddr(...)
```

_Originally posted by @ggoranov in https://github.com/dusk-network/dusk-blockchain/pull/194_

**Comment by CPerezz** [comment]
Moving to UDP will mean that we will have data losses in block_data transfers since as @toghrulmaharramov suggested, blocks will have an approximate size of 5MB. And that is quite huge to assume, since 5MB without any tx/rx errors it's something quite unlikely to happen.

We could implement or use CRC codes in order to mitigate the data lose and have a decent error correction rate, but it will probably decrease the performance.

**Comment by CPerezz** [comment]
Also, I forgot to mention that the code should already be doing this.
Check: https://github.com/dusk-network/dusk-blockchain/blob/ffb61f2bf7580bac17c841ce75dbe95f5de77ef3/pkg/p2p/kadcast/network.go#L34

Since once you try to read, if the deadline pops, you will catch the error and open a new listening process with a fresh deadline.

This is also implemented in the TCP listener I implemented in #194
